lib/ts/recipe/session/utils.ts

```diff
@@ -93,12 +108,18 @@ export function validateAndNormaliseUserInput(
             ? undefined
             : normaliseSessionScopeOrThrowError(recipeInstance.getRecipeId(), config.cookieDomain);
 
-    let cookieSameSite =
+    let topLevelAPIDomain = getTopLevelDomain(appInfo.apiDomain.getAsStringDangerous());
+    let topLevelWebsiteDomain = getTopLevelDomain(appInfo.websiteDomain.getAsStringDangerous());
+
+    let cookieSameSite: "strict" | "lax" | "none" =
+        topLevelAPIDomain !== null && topLevelAPIDomain !== topLevelWebsiteDomain ? "none" : "lax";
+    cookieSameSite =
         config === undefined || config.cookieSameSite === undefined
-            ? "lax"
+            ? cookieSameSite
             : normaliseSameSiteOrThrowError(recipeInstance.getRecipeId(), config.cookieSameSite);
 
-    let cookieSecure = config === undefined || config.cookieSecure === undefined ? false : config.cookieSecure;
+    let cookieSecure = appInfo.apiDomain.getAsStringDangerous().startsWith("https") ? true : false;
```

Comment:
Can change this to `let cookieSecure = appInfo.apiDomain.getAsStringDangerous().startsWith("https")`